tools/get_dicom.pl

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/perl -w 
+use strict;
+
+=pod
+
+=head1 NAME
+
+get_dicom.pl - extracts DICOM files for specific patient names/scan types
+
+=head1 SYNOPSIS
+
+perl get_dicom.pl [-p profile] [-n patient_name_patterns] [-t scan_type_patterns] [-d tmp_dir] [-o tarBasename]
+
+Available options are:
+
+-p      : name of the config file in C<../dicom-archive/.loris_mri> (defaults to
+          prod).
+-n      : comma separated list of MySQL patterns for the patient names that a DICOM file
+          has to have in order to be extracted. A DICOM file only has to match one of the 
+          patterns to be extracted. If no pattern is specified, then the patient name is 
+          not used to determine which DICOM files to extract.
+-t      : comma separated list of MySQL patterns of the acquisition protocols (scan types
+          names) that a DICOM file has to have in order to be extracted. A DICOM file only
+          has to match one of the patterns to be extracted. If no pattern is specified, then
+          the scan type name is not used to determine which DICOM files to extract.
+-d      : extract the files in directory <dir_argument>/get_dicom.pl.<UNIX_process_number>
+          For example with -d /data/tmp, the DICOM files will be extracted in 
+          /data/tmp/get_dicom.pl.67888 (assuming 67888 is the process number). By default, dir_argument
+          is set to /tmp.
+-o      : basename of the final tar.gz file to produce, in the current directory (defaults to dicoms.tar.gz)
+
+=head1 DESCRIPTION
+
+This script first connects to the database to build the list of tarchives for which
+the patient names match the list of patterns specified as argument. The script will then examine 
+these tarchives and look for the MINC files whose scan types (acquisition protocol names) match 
+the list of patterns passed as argument. It then extracts the DICOM files associated to each
+MINC file and writes them in the extraction directory (see -d option), in a subdirectory with name
+
+<pscid>/<visit_label>/<acquisition_date>/<protocol>_<minc_index>
+
+where <minc_index> is the index number of the MINC file to which the DICOMs are associated.
+Finally, a tar is created which contains all the DICOM files that were extract
+
+=head2 Methods
+
+=cut
+
+use NeuroDB::DBI;
+use NeuroDB::ExitCodes;
+
+use File::Path;
+use File::Basename;
+use Getopt::Tabular;
+
+#-----------------------------------------------#
+# Triggers the deletion of the tmp extract dir  #
+# if something like CTRL-C is pressed during    #
+# execution of the script                       #
+#-----------------------------------------------#
+use sigtrap 'handler' => \&rmTmpExtractDir, 'normal-signals';
+
+my $patientNames;
+my $scanTypes;
+my $profile           = 'prod';
+my $tmpExtractBaseDir = '/tmp';
```

Comment:
here I would use the bash variable `$TMPDIR` that is set in the `environment` file instead of hardcoding it to `/tmp`

Reply:
Done.